scripts/mkchangelog

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+
+import datetime
+import json
+import logging
+import os
+import pathlib
+import re
+import subprocess
+import tempfile
+
+
+logger = logging.getLogger(__name__)
+
+
+def gh(*args, **kwargs):
+    return subprocess.run(
+        ["gh", *args],
+        check=True,
+        env={"GH_TOKEN": os.getenv("GH_TOKEN", "")},
+        **kwargs,
+    )
+
+
+def list_merged_pull_requests():
+    logger.info("Reading pull requests from repository.")
+    result = gh(
+        "api",
+        "/repos/gip-inclusion/les-emplois/pulls?state=closed",
+        "--header",
+        "Accept: application/vnd.github+json",
+        "--header",
+        "X-GitHub-Api-Version: 2022-11-28",
+        "--paginate",
+        capture_output=True,
+    )
+    return result.stdout
+
+
+def open_pull_request(branch):
+    gh("pr", "create", "--fill", "--label", "no-changelog", "--base", "master", "--head", branch)
+
+
+def main():
+    changelog_path = pathlib.Path("CHANGELOG.md")
+    with (
+        open(changelog_path, "r+") as changelog,
+        tempfile.NamedTemporaryFile(dir=os.getcwd(), mode="w+", suffix="~", delete=False) as new_changelog,
+    ):
+        try:
+            for line in changelog:
+                if line.strip() in ["# Journal des modifications", ""]:
+                    new_changelog.write(line)
+                    continue
+                break
+            last_entry_header = line
+
+            m = re.match(r"## \[(?P<sprint>[0-9]+)\] - (?P<date>[0-9]+-[0-9]+-[0-9]+)", last_entry_header)
+            sprint_number = int(m.group("sprint")) + 1
+            sprint_start = datetime.date.fromisoformat(m.group("date"))
+            sprint_end = sprint_start + datetime.timedelta(days=14)
```

Comment:
:grimacing: 
Je me demande si on ne devrait finalement pas passer par une action manuelle (soit un script à lancer à la main comme tu le suggérais initialement :see_no_evil: ) soit https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#defining-inputs-for-manually-triggered-workflows où on indiquerait juste
`Sprint Id, End date`.
Ou alors se baser sur les objets Releases: cela permettrait de créer les releases à l'avance en indiquant:
`[85] - 2024-04-22`
`[86] - 2024-05-13` pour indiquer que le sprint 86 va durer 3 semaines ?

Reply:
C’est dommage d’exploiter les objets Releases sans utiliser la fonctionnalité de changelog auto de GitHub :shrug:.

Le code me semble facile à modifier si on veut d’un sprint qui dure plus X semaines. L’intéret que ce soit automatique est que dans la plupart des sprints, on n’aura qu’à relire la PR.

Après, une action manuelle ne me dérange pas énormément, on pourrait ne lui donner que la date de fin. Le numéro du sprint est simplement un incrément, je ne vois pas bien ce qui viendrait le casser ?

Reply:
Est-ce que justement ce ne serait pas intéressant de se reposer sur les releases GH ?

`sprint_number` = `latest release number + 1`
`sprint_start` = `latest release date + 1`
`sprint_end` = `now`

Et effectivement trigger par une action manuelle.
Le contenu de la release GH pourrait être le même que ce que l'on prepend au changelog

Reply:
Autre alternative (à valider avec le métier :sweat_smile: ): on passe à un changelog automatique par semaine avec comme nom le numéro de semaine (2024-16 actuellement).
A la fin de notre sprint il suffit de présenter le changelog des X dernières semaines.

Vu qu'on est en déploiement continu le changelog par sprint n'a pas vraiment de logique de toutes façons :angel: .

Reply:
https://itou-inclusion.slack.com/archives/C01AQKD7MAN/p1713446635845979